stat.go

```diff
@@ -125,6 +125,35 @@ func ChiSquare(obs, exp []float64) float64 {
 	return result
 }
 
+// CircularMean returns the circular mean of the dataset.
+//	atan2(\sum_i w_i * sin(alpha_i), \sum_i w_i * cos(alpha_i))
+// If weights is nil then all of the weights are 1. If weights is not nil, then
+// len(x) must equal len(weights).
+func CircularMean(x, weights []float64) float64 {
+	if weights != nil && len(x) != len(weights) {
+		panic("stat: slice length mismatch")
+	}
+
+	var aX, aY float64
+	if weights != nil {
+		for i := range x {
+			aX += weights[i] * math.Cos(x[i])
```

Comment:
There used to be a performance difference, but I don't know if the compiler has gotten smart enough in the last few Go versions to optimize.  Regardless of performance, [Effective Go](https://golang.org/doc/effective_go.html#for) says to use it:
> If you only need the second item in the range (the value), use the blank identifier, an underscore, to discard the first:
> ```
sum := 0
for _, value := range array {
    sum += value
}
```
So it is idiomatic and very slightly less complicated to use `_, v`.  If it occurs in other places in stat.go (or any gonum package) then that is a mistake and we should fix it.